### PR TITLE
fix(lerobot_local): canonicalize bare camera frames on the declarative embodiment path

### PR DESCRIPTION
**Pull request by cagataycali**
## Problem

Driving a **single-camera** lerobot checkpoint (e.g. an ACT policy declaring only `observation.images.front`) through the declarative `embodiment` path crashes, while the legacy (no-embodiment) path works.

On the declarative path the camera rename (`front` -> `observation.images.front`) happens **inside** the preprocessor pipeline, so the observation handed to `_canonicalize_obs_images` still carries the **bare** source key. That helper gated purely on the `"image"` substring, so a bare `front` frame was skipped and reached lerobot's normalizer as raw HWC uint8:

```
Preprocessor pipeline failed: value cannot be converted to type uint8 without overflow
```

(or a `torch.stack ... [3, H] and [H]` shape mismatch). `_canonicalize_obs_images`'s own docstring claims it "runs before BOTH the embodiment and legacy preprocess paths", but the substring gate made it a no-op for bare keys on the declarative branch.

## Change

- Add `_embodiment_image_source_keys()` — the embodiment `obs_rename` sources whose target is a declared image (VISUAL) input feature (resolved via the existing `_declared_feature_is_image`, so VISUAL keys without an `image` substring are honored too).
- `_canonicalize_obs_images(observation, image_source_keys=None)` now canonicalizes a bare key when it is one of those image sources. The declarative branch in `get_actions` passes the embodiment's source keys; the legacy `"image"`-substring-only behavior is unchanged when `image_source_keys` is `None`.

No public API change; the fix is confined to the lerobot_local observation-canonicalization path.

## Tests

New `tests/policies/lerobot_local/test_declarative_obs_image_canonicalization.py`. The behavioral case drives `get_actions` through the declarative branch with a bare `front` HWC uint8 frame and asserts it reaches `preprocess` as CHW `float32` in `[0,1]`. **Pre-fix: 4 of 5 fail** (bare frame reaches `preprocess` as raw HWC uint8); the 5th is a control pinning that the legacy substring-only path is unchanged. Post-fix all pass, and the existing `test_policy.py` / `test_camera_obs_rename_preflight.py` / `test_configure_embodiment.py` suites stay green (178 passed).

Gate: `ruff check` + `ruff format --check` + `mypy strands_robots tests tests_integ` clean; tests run with `MUJOCO_GL=egl`.

## Rollout in MuJoCo sim (headless)

SmolVLA on the SO-101 through the `lerobot_local` pipeline this change lives in — headless MuJoCo (`MUJOCO_GL=egl`), 60 steps, recorded to a LeRobotDataset (3 cameras, per-camera MP4s, 60 frames verified on reopen) with continuous motion (no frozen frames):

smolvla so101 rollout

Docs: `docs/policies/camera-naming.md` gains a short note that bare camera keys are canonicalized on the declarative path (no `image`-substring naming requirement).

### CODE REVIEW

**Comment by cagataycali** [comment]
Addresses **F3** of #880 (the declarative path could not process a single-camera observation: a bare camera key skipped canonicalization and overflowed the normalizer). F2 (the cuda-pinned `device_processor`) was already fixed in #883; F1 (a single-camera SO embodiment + `obs_rename` key removal) remains open for a follow-up.

**Review by yinsong1986 — APPROVED** [review]
## Summary

Fixes a crash on the declarative `embodiment` path of `LerobotLocalPolicy` for single-camera (and bare-keyed) checkpoints. On that path the camera rename (`front` -> `observation.images.front`) runs inside the preprocessor pipeline, so the observation reaching `_canonicalize_obs_images` still carries the bare source key. The old `"image" in key` substring gate skipped it, and the raw HWC uint8 frame reached lerobot's normalizer and overflowed. The PR adds `_embodiment_image_source_keys()` (the `obs_rename` sources whose target is a declared VISUAL feature, resolved via the existing `_declared_feature_is_image`) and threads them into `_canonicalize_obs_images(..., image_source_keys=...)`, so bare camera keys are canonicalized to CHW float32 [0,1] on the declarative branch. The diff matches the description.

## What's good

- Additive and tightly scoped: both new methods are `_`-prefixed and not exported; no public API, wire-format, persisted-schema, env-var, or CLI surface changes.
- Legacy behavior is exactly preserved when `image_source_keys` is `None` (the substring-only gate), pinned by `test_bare_camera_key_untouched_without_source_keys`.
- Reuses the authoritative `_declared_feature_is_image` (VISUAL type metadata, not a name heuristic), so it stays consistent with the prior bare-key routing fix.
- No crash surface in the new code: `_input_features` is always a dict (initialized `{}` at policy.py:285) so `.get(dst)` is safe, and `_embodiment is None` is guarded before iterating `obs_rename`.
- Plain-ASCII docstrings/strings, no emojis; regression tests cover the source-key derivation, the canonicalization, the legacy no-op, and the end-to-end declarative branch.

## Verification suggestions

Standard CI (`hatch run test` on the lerobot_local suite) plus the in-PR headless MuJoCo SmolVLA/SO-101 rollout cover this. If spot-checking, confirm no double-`/255` against a checkpoint whose pipeline already includes an image-rescale step (a single-cam ACT producing valid actions is the right signal).
